packages/ui/src/users/UsersList.tsx

```diff
@@ -121,13 +125,33 @@ const UsersList: FunctionComponent = () => {
 
   const generateDefaultUser = (): UserWithOrg => {
     return {
-      ...(new UserWithOrg()),
+      ...new UserWithOrg(),
       permissions: [
         PERMISSIONS.CAN_MANAGE_DOCUMENTS,
         PERMISSIONS.CAN_VIEW_DOCUMENTS,
-      ]
+      ],
+    };
+  };
+
+  const onUserDelete = async (user: UserWithOrg) => {
+    try {
+      setState({
```

Comment:
This got me confused at first. It seems to me like the gateway contains a `useMergeState` hook to simplify state with hooks. I'm guessing this is so you don't need to have multiple `const [someState, setSomeState] = useState(...)` lines I suppose?

Reply:
(it's confusing because it exactly mirrors the old React `setState()` methods.

Reply:
Also, not saying this as criticism, I'm just curious about the reasoning for this ;-)

Reply:
Yea that is the idea and I think setState is a good name. Do you see any problem having the same name as the Class component method?

Reply:
As I remember I was just for that because it is easy to remember and understand if you know what this.setState does

Reply:
Makes sense, thanks for the explanation!